Cannot find the VDC storage policy with v0.10

Hi,

After updating 0.9.6 to v0.10, I can no longer create VMs on vCloud
```
* vcd_vapp.terravm: Error creating vapp: &errors.errorString{s:"Error creating vapp: &errors.errorString{s:\"error instantiating a new vApp: API Error: 400: [ e172bcb4-db68-49b4-bd87-8d8d04a95adb ] Cannot find the VDC storage policy \\\"545fb858-3a90-4801-86d4-00d275ae1b6d\\\".\"}"}
```

Switching back to terraform 0.9.6, the same main.tf works fine.

There is no storage reference in the TF file, and its not complicated:
```
provider "vcd" {
  user                 = "${var.username}"
  password             = "${var.password}"
  org                  = "${var.org}"
  url                  = "${var.url}"
  vdc                  = "${var.vdc}"
  maxRetryTimeout      = "${var.maxRetryTimeout}"
  allow_unverified_ssl = "${var.allow_unverified_ssl}"
}
resource "vcd_vapp" "terravm" {
  name          = "${var.vmname}"
  catalog_name  = "${var.catalog}"
  template_name = "${var.template}"
  memory        = "${var.memory}"
  cpus          = "${var.cpus}"
  metadata {
    owner       = "${var.owner}"
    description = "${var.description}"
  }
  network_name = "${var.network}"
  ip = "${var.ip}"
..
```
Any ideas?

@bobbydeveaux 

If I try this
```
storage_profile_reference := nil

// Override default_storage_profile if we find the given storage profile
if d.Get("storage_profile").(string) != "" {
	storage_profile_reference, err = vcdClient.OrgVdc.FindStorageProfileReference(d.Get("storage_profile").(string))
	if err != nil {
		log.Printf("Error assigning custom storage profile. Using default storage_profile.")
	}
} 
```
I get compile errors:
```
==> Checking that code complies with gofmt requirements...
go install
# github.com/terraform-providers/terraform-provider-vcd/vcd
vcd/resource_vcd_vapp.go:121: use of untyped nil
make: *** [GNUmakefile:7: build] Error 2
```
And if I try to use an empty types.Reference{} it fails to create a vapp due to this restriction:
```
type Reference struct {
	HREF string `xml:"href,attr"`
	ID   string `xml:"id,attr,omitempty"`
	Type string `xml:"type,attr,omitempty"`
	Name string `xml:"name,attr,omitempty"`
}
```
HREF for types.Reference structs can't be empty.

--- a/vcd/resource_vcd_vapp.go
+++ b/vcd/resource_vcd_vapp.go
@@ -117,25 +117,12 @@ func resourceVcdVAppCreate(d *schema.ResourceData, meta interface{}) error {
 
 			storage_profile_reference := types.Reference{}
 
-			// Use default_storage_profile as fallback
-			storage_profile_query_params := make(map[string]string)
-			storage_profile_query_params["type"] = "orgVdcStorageProfile"
-			storage_profile_query_params["format"] = "records"
-			storage_profiles, err := vcdClient.VCDClient.Query(storage_profile_query_params)
-			default_storage_profile, err := vcdClient.OrgVdc.GetDefaultStorageProfileReference(storage_profiles.Results)
-			if err != nil {
-				return fmt.Errorf("Couldn't find storage_profile and no default storage_profile avaiable %s", err)
-			}
-
 			// Override default_storage_profile if we find the given storage profile
 			if d.Get("storage_profile").(string) != "" {
 				storage_profile_reference, err = vcdClient.OrgVdc.FindStorageProfileReference(d.Get("storage_profile").(string))
 				if err != nil {
-					storage_profile_reference = default_storage_profile
-					log.Printf("Using default storage_profile: %s", storage_profile_reference)
+					return fmt.Errorf("Error finding storage profile %s", d.Get("storage_profile").(string))
 				}
-			} else {
-				storage_profile_reference = default_storage_profile
 			}
 
 			log.Printf("storage_profile %s", storage_profile_reference)

--- a/vcd/resource_vcd_vapp_test.go
+++ b/vcd/resource_vcd_vapp_test.go
@@ -19,7 +19,7 @@ func TestAccVcdVApp_PowerOff(t *testing.T) {
 		CheckDestroy: testAccCheckVcdVAppDestroy,
 		Steps: []resource.TestStep{
 			resource.TestStep{
-				Config: fmt.Sprintf(testAccCheckVcdVApp_basic, os.Getenv("VCD_EDGE_GATEWAY")),
+				Config: fmt.Sprintf(testAccCheckVcdVApp_basic, os.Getenv("VCD_EDGE_GATEWAY"), os.Getenv("VCD_EDGE_GATEWAY")),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckVcdVAppExists("vcd_vapp.foobar", &vapp),
 					testAccCheckVcdVAppAttributes(&vapp),
@@ -33,7 +33,7 @@ func TestAccVcdVApp_PowerOff(t *testing.T) {
 			},
 
 			resource.TestStep{
-				Config: fmt.Sprintf(testAccCheckVcdVApp_basic, os.Getenv("VCD_EDGE_GATEWAY")),
+				Config: fmt.Sprintf(testAccCheckVcdVApp_basic, os.Getenv("VCD_EDGE_GATEWAY"), os.Getenv("VCD_EDGE_GATEWAY")),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr(
 						"vcd_vapp.foobar_allocated", "name", "foobar-allocated"),

